p2p/simulations/adapters/exec.go

```diff
@@ -137,34 +157,70 @@ func (n *ExecNode) Start(snapshots map[string][]byte) (err error) {
 	// encode a copy of the config containing the snapshot
 	confCopy := *n.Config
 	confCopy.Snapshots = snapshots
+	confCopy.PeerAddrs = make(map[string]string)
+	for id, node := range n.adapter.nodes {
+		confCopy.PeerAddrs[id.String()] = node.wsAddr
+	}
 	confData, err := json.Marshal(confCopy)
 	if err != nil {
 		return fmt.Errorf("error generating node config: %s", err)
 	}
 
-	// create a net.Pipe for RPC communication over stdin / stdout
-	pipe1, pipe2 := net.Pipe()
+	// use a pipe for stderr so we can both copy the node's stderr to
+	// os.Stderr and read the WebSocket address from the logs
+	stderrR, stderrW := io.Pipe()
+	stderr := io.MultiWriter(os.Stderr, stderrW)
 
 	// start the node
 	cmd := n.newCmd()
-	cmd.Stdin = pipe1
-	cmd.Stdout = pipe1
-	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = stderr
 	cmd.Env = append(os.Environ(), fmt.Sprintf("_P2P_NODE_CONFIG=%s", confData))
 	if err := cmd.Start(); err != nil {
 		return fmt.Errorf("error starting node: %s", err)
 	}
 	n.Cmd = cmd
 
+	// read the WebSocket address from the stderr logs
+	var wsAddr string
+	errC := make(chan error)
+	go func() {
+		s := bufio.NewScanner(stderrR)
```

Comment:
It does, but we set it explicitly in [`execP2PNode`](https://github.com/ethersphere/go-ethereum/blob/network-testing-framework/p2p/simulations/adapters/exec.go#L264).